src/grpc_server.cc

```diff
@@ -2290,11 +2290,7 @@ class InferHandlerState {
   ~InferHandlerState()
   {
 #ifdef TRITON_ENABLE_TRACING
-    if (trace_ != nullptr) {
-      for (const auto& timestamp : trace_timestamps_) {
-        trace_->CaptureTimestamp(timestamp.first, timestamp.second);
-      }
-    }
```

Comment:
This didn't call trace_.reset() or trace_timestamps_.clear() before as they would automatically be cleaned up, but I figured it was cleaner to make use of the same helper here anyways.